src/api/__tests__/hooks.unit.test.ts.tsx

```diff
@@ -1,533 +1,315 @@
-  });
-});
-
-describe("useApiGetPaginated", () => {
-  afterEach(() => {
-    mockAxios.reset();
-  });
-
-  it("calls axios.get (sanity check)", () => {
-    renderHook(() => h.useApiGetPaginated("test"));
-
-    expect(mockAxios.get.mock.calls[0][0]).toBe("test");
-  });
-
-  it("returns an object with null values when request is not ready", () => {
-    const { result } = renderHook(() => h.useApiGetPaginated("test"));
-
-    const [{ next, previous, page }, err] = result.current;
-
-    expect(next).toBeNull();
-    expect(previous).toBeNull();
-    expect(page).toBeNull();
-    expect(err).toBeUndefined();
+    expect(error).toEqual(MOCK_ERROR.error);
   });
 
-  it("returns pagination functions that paginate to next/previous page", async () => {
-    const { result } = renderHook(() => h.useApiGetPaginated("test"));
-    const data = {
-      count: 20,
-      page: 2,
-      num_pages: 2,
-      next: "test1",
-      previous: "test2",
-      results: []
-    };
-
-    act(() => {
-      mockAxios.mockResponse({
-        data: {
-          success: true,
-          data
-        }
-      });
-    });
-
-    const [resp1, err1] = result.current;
-    expect(resp1.next).not.toBeNull();
-    expect(resp1.previous).not.toBeNull();
-    expect(resp1.page).toEqual(data);
-    expect(err1).toBeUndefined();
+  it("correctly updates RequestState", async () => {
+    const get = retrieveFactory<string>("bingo/");
+    const fn = () => get({ id: "15" });
+    const { result } = renderHook(() => useAPI(fn));
 
-    act(() => {
-      resp1.next!();
-    });
-
-    expect(mockAxios.get.mock.calls[1][0]).toBe("test1");
+    let [resp, error, info] = result.current;
+    expect(info.state).toBe(RequestState.Running);
 
-    act(() => {
-      mockAxios.mockResponse({
-        data: {
-          success: true,
-          data
+    await act(async () => {
+      mockAxios.mockResponseFor(
+        {
+          method: "GET",
+          url: "bingo/15/",
+        },
+        {
+          data: MOCK_RESPONSE,
         }
-      });
+      );
     });
 
-    const [resp2, err2] = result.current;
-    expect(resp2.next).not.toBeNull();
-    expect(resp2.previous).not.toBeNull();
-    expect(resp2.page).toEqual(data);
-    expect(err2).toBeUndefined();
-
-    act(() => {
-      resp2.previous!();
-    });
-
-    expect(mockAxios.get.mock.calls[2][0]).toBe("test2");
+    [resp, error, info] = result.current;
+    expect(info.state).toBe(RequestState.Complete);
   });
 
-  it("returns a null next paginator if next page does not exist", () => {
-    const { result } = renderHook(() => h.useApiGetPaginated("test"));
-    const data = {
-      count: 20,
-      page: 2,
-      num_pages: 2,
-      next: null,
-      previous: "test",
-      results: []
-    };
-
-    act(() => {
-      mockAxios.mockResponse({
-        data: {
-          success: true,
-          data
-        }
-      });
+  it("correctly updates return value on multiple fetches", async () => {
```

Comment:
Test case should be renamed to "clears error state after subsequent successful fetches"

Reply:
https://github.com/UWmathNEWS/slugline-web/pull/33/commits/d19d9dec491610ceb2c614df655032bdf3136299